packages/protocol/contracts/governance/Governance.sol

```diff
@@ -486,8 +486,12 @@ contract Governance is
   {
     Proposals.Proposal storage proposal = proposals[proposalId];
     require(_isDequeuedProposal(proposal, proposalId, index), "Proposal not dequeued");
-    Proposals.Stage stage = proposal.getDequeuedStage(stageDurations);
+    Proposals.Stage stage = proposal.getDequeuedStage(stageDurations, true);
     if (_isDequeuedProposalExpired(proposal, stage)) {
+      if (proposal.transactions.length == 0) {
+        // mark proposals with no transactions as executed
+        emit ProposalExecuted(proposalId);
```

Comment:
Will this only happen for proposals that were approved and passed referendum?

Reply:
Fixed